[Bug] CKAN raises unhandled exception when downloading large-*ish* mods on Manjaro Linux

Background
----------
* Operating  System: Manjaro Linux kernel 4.19.78-1-MANJARO
* CKAN Version: 1.26.4
* KSP Version: 1.8.0

Problem
-------
When downloading large-ish mods, on my ADSL internet connection, after an unkown number of seconds which i suspect could be precisely measured and reproductible, CKAN crashes, vomiting a huge stacktrace. Upon investigation, seems like it's a TCP timeout issue, altough the download progresses (and i can indeed download the incriminated file outside CKAN).

Of course, this exception should not be raised, since the download is progressing at a steady pace.


**Steps to reproduce**
* Go to my house, or any house with slow internet (1MB/s or so)
* Install CKAN via the [AUR](https://aur.archlinux.org/packages/ckan/) (UPDATE: running a downloaded ckan.exe gives the same results)
* Click on any large mod (>= 100MB), then proceed to installation, wait for what feels like 1 minute...
* See error, if you launched CKAN in a terminal. Otherwise, nothing.

**Expected behavior**
Of course, i would expect CKAN to continue downloading. However, I suspect the issue is more on mono's side, but i'm far from even begining to understand C#'s ecosystem...

You will find the complete shell session below:
```
~ $ ckan
Could not set X locale modifiers
133303 [1] ERROR CKAN.ErrorDialog (null) - System.Net.WebException: The operation has timed out.
  at System.Net.HttpWebRequest.RunWithTimeoutWorker[T] (System.Threading.Tasks.Task`1[TResult] workerTask, System.Int32 timeout, System.Action abort, System.Func`1[TResult] aborted, System.Threading.CancellationTokenSource cts) [0x000f8] in <883cd000faa842148a99b174e3bd5ca4>:0 
  at System.Net.HttpWebRequest.EndGetResponse (System.IAsyncResult asyncResult) [0x00020] in <883cd000faa842148a99b174e3bd5ca4>:0 
  at System.Net.WebClient.GetWebResponse (System.Net.WebRequest request, System.IAsyncResult result) [0x00000] in <883cd000faa842148a99b174e3bd5ca4>:0 
  at System.Net.WebClient.GetWebResponseTaskAsync (System.Net.WebRequest request) [0x0008d] in <883cd000faa842148a99b174e3bd5ca4>:0 
  at System.Net.WebClient.DownloadBitsAsync (System.Net.WebRequest request, System.IO.Stream writeStream, System.ComponentModel.AsyncOperation asyncOp, System.Action`3[T1,T2,T3] completionDelegate) [0x0008d] in <883cd000faa842148a99b174e3bd5ca4>:0 
133340 [1] ERROR CKAN.ErrorDialog (null) - System.Net.WebException: The operation has timed out.
  at System.Net.HttpWebRequest.RunWithTimeoutWorker[T] (System.Threading.Tasks.Task`1[TResult] workerTask, System.Int32 timeout, System.Action abort, System.Func`1[TResult] aborted, System.Threading.CancellationTokenSource cts) [0x000f8] in <883cd000faa842148a99b174e3bd5ca4>:0 
  at System.Net.HttpWebRequest.EndGetResponse (System.IAsyncResult asyncResult) [0x00020] in <883cd000faa842148a99b174e3bd5ca4>:0 
  at System.Net.WebClient.GetWebResponse (System.Net.WebRequest request, System.IAsyncResult result) [0x00000] in <883cd000faa842148a99b174e3bd5ca4>:0 
  at System.Net.WebClient.GetWebResponseTaskAsync (System.Net.WebRequest request) [0x0008d] in <883cd000faa842148a99b174e3bd5ca4>:0 
  at System.Net.WebClient.DownloadBitsAsync (System.Net.WebRequest request, System.IO.Stream writeStream, System.ComponentModel.AsyncOperation asyncOp, System.Action`3[T1,T2,T3] completionDelegate) [0x0008d] in <883cd000faa842148a99b174e3bd5ca4>:0 
133341 [1] ERROR CKAN.ErrorDialog (null) - System.Net.WebException: The operation has timed out.
  at System.Net.HttpWebRequest.RunWithTimeoutWorker[T] (System.Threading.Tasks.Task`1[TResult] workerTask, System.Int32 timeout, System.Action abort, System.Func`1[TResult] aborted, System.Threading.CancellationTokenSource cts) [0x000f8] in <883cd000faa842148a99b174e3bd5ca4>:0 
  at System.Net.HttpWebRequest.EndGetResponse (System.IAsyncResult asyncResult) [0x00020] in <883cd000faa842148a99b174e3bd5ca4>:0 
  at System.Net.WebClient.GetWebResponse (System.Net.WebRequest request, System.IAsyncResult result) [0x00000] in <883cd000faa842148a99b174e3bd5ca4>:0 
  at System.Net.WebClient.GetWebResponseTaskAsync (System.Net.WebRequest request) [0x0008d] in <883cd000faa842148a99b174e3bd5ca4>:0 
  at System.Net.WebClient.DownloadBitsAsync (System.Net.WebRequest request, System.IO.Stream writeStream, System.ComponentModel.AsyncOperation asyncOp, System.Action`3[T1,T2,T3] completionDelegate) [0x0008d] in <883cd000faa842148a99b174e3bd5ca4>:0 
133341 [1] ERROR CKAN.ErrorDialog (null) - System.Net.WebException: The operation has timed out.
  at System.Net.HttpWebRequest.RunWithTimeoutWorker[T] (System.Threading.Tasks.Task`1[TResult] workerTask, System.Int32 timeout, System.Action abort, System.Func`1[TResult] aborted, System.Threading.CancellationTokenSource cts) [0x000f8] in <883cd000faa842148a99b174e3bd5ca4>:0 
  at System.Net.HttpWebRequest.EndGetResponse (System.IAsyncResult asyncResult) [0x00020] in <883cd000faa842148a99b174e3bd5ca4>:0 
  at System.Net.WebClient.GetWebResponse (System.Net.WebRequest request, System.IAsyncResult result) [0x00000] in <883cd000faa842148a99b174e3bd5ca4>:0 
  at System.Net.WebClient.GetWebResponseTaskAsync (System.Net.WebRequest request) [0x0008d] in <883cd000faa842148a99b174e3bd5ca4>:0 
  at System.Net.WebClient.DownloadBitsAsync (System.Net.WebRequest request, System.IO.Stream writeStream, System.ComponentModel.AsyncOperation asyncOp, System.Action`3[T1,T2,T3] completionDelegate) [0x0008d] in <883cd000faa842148a99b174e3bd5ca4>:0 
133341 [1] ERROR CKAN.ErrorDialog (null) - System.Net.WebException: The operation has timed out.
  at System.Net.HttpWebRequest.RunWithTimeoutWorker[T] (System.Threading.Tasks.Task`1[TResult] workerTask, System.Int32 timeout, System.Action abort, System.Func`1[TResult] aborted, System.Threading.CancellationTokenSource cts) [0x000f8] in <883cd000faa842148a99b174e3bd5ca4>:0 
  at System.Net.HttpWebRequest.EndGetResponse (System.IAsyncResult asyncResult) [0x00020] in <883cd000faa842148a99b174e3bd5ca4>:0 
  at System.Net.WebClient.GetWebResponse (System.Net.WebRequest request, System.IAsyncResult result) [0x00000] in <883cd000faa842148a99b174e3bd5ca4>:0 
  at System.Net.WebClient.GetWebResponseTaskAsync (System.Net.WebRequest request) [0x0008d] in <883cd000faa842148a99b174e3bd5ca4>:0 
  at System.Net.WebClient.DownloadBitsAsync (System.Net.WebRequest request, System.IO.Stream writeStream, System.ComponentModel.AsyncOperation asyncOp, System.Action`3[T1,T2,T3] completionDelegate) [0x0008d] in <883cd000faa842148a99b174e3bd5ca4>:0 
133353 [1] ERROR CKAN.ErrorDialog (null) - Unhandled exception:
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.InvalidOperationException: The form is already displayed as a modal dialog.
  at System.Windows.Forms.Form.ShowDialog (System.Windows.Forms.IWin32Window owner) [0x00081] in <f73fa304d089422f9e23b25394111843>:0 
  at System.Windows.Forms.Form.ShowDialog () [0x00000] in <f73fa304d089422f9e23b25394111843>:0 
  at CKAN.ErrorDialog.ShowErrorDialog (System.String text, System.Object[] args) [0x00071] in <ea98fd2118b84fc7ad1a2141a259ca95>:0 
  at CKAN.ErrorDialog+<>c__DisplayClass2_0.<ShowErrorDialog>b__0 () [0x00000] in <ea98fd2118b84fc7ad1a2141a259ca95>:0 
  at (wrapper managed-to-native) System.Reflection.RuntimeMethodInfo.InternalInvoke(System.Reflection.RuntimeMethodInfo,object,object[],System.Exception&)
  at System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x0006a] in <4f52a2659e1844fea7ab38854dcbcf68>:0 
   --- End of inner exception stack trace ---
  at System.Windows.Forms.AsyncMethodResult.EndInvoke () [0x0004c] in <f73fa304d089422f9e23b25394111843>:0 
  at System.Windows.Forms.Control.EndInvoke (System.IAsyncResult asyncResult) [0x00000] in <f73fa304d089422f9e23b25394111843>:0 
  at (wrapper remoting-invoke-with-check) System.Windows.Forms.Control.EndInvoke(System.IAsyncResult)
  at System.Windows.Forms.Control.Invoke (System.Delegate method, System.Object[] args) [0x00021] in <f73fa304d089422f9e23b25394111843>:0 
  at (wrapper remoting-invoke-with-check) System.Windows.Forms.Control.Invoke(System.Delegate,object[])
  at System.Windows.Forms.Control.Invoke (System.Delegate method) [0x0001d] in <f73fa304d089422f9e23b25394111843>:0 
  at (wrapper remoting-invoke-with-check) System.Windows.Forms.Control.Invoke(System.Delegate)
  at CKAN.ErrorDialog.ShowErrorDialog (System.String text, System.Object[] args) [0x0002c] in <ea98fd2118b84fc7ad1a2141a259ca95>:0 
  at (wrapper remoting-invoke-with-check) CKAN.ErrorDialog.ShowErrorDialog(string,object[])
  at CKAN.Main.ErrorDialog (System.String text, System.Object[] args) [0x0000d] in <ea98fd2118b84fc7ad1a2141a259ca95>:0 
  at (wrapper remoting-invoke-with-check) CKAN.Main.ErrorDialog(string,object[])
  at CKAN.GUIUser.RaiseError (System.String message, System.Object[] args) [0x00005] in <ea98fd2118b84fc7ad1a2141a259ca95>:0 
  at CKAN.NetAsyncModulesDownloader.ModuleDownloadComplete (System.Uri url, System.String filename, System.Exception error) [0x0001c] in <ea98fd2118b84fc7ad1a2141a259ca95>:0 
  at CKAN.NetAsyncDownloader.FileDownloadComplete (System.Int32 index, System.Exception error) [0x00121] in <ea98fd2118b84fc7ad1a2141a259ca95>:0 
  at CKAN.NetAsyncDownloader+<>c__DisplayClass15_0.<DownloadNative>b__1 (System.Object sender, System.ComponentModel.AsyncCompletedEventArgs args) [0x00012] in <ea98fd2118b84fc7ad1a2141a259ca95>:0 
  at CKAN.NetAsyncDownloader+NetAsyncDownloaderDownloadPart.<ResetAgent>b__21_1 (System.Object sender, System.ComponentModel.AsyncCompletedEventArgs args) [0x00008] in <ea98fd2118b84fc7ad1a2141a259ca95>:0 
  at System.Net.WebClient.OnDownloadFileCompleted (System.ComponentModel.AsyncCompletedEventArgs e) [0x0000a] in <883cd000faa842148a99b174e3bd5ca4>:0 
  at System.Net.WebClient.<StartAsyncOperation>b__78_4 (System.Object arg) [0x00000] in <883cd000faa842148a99b174e3bd5ca4>:0 
  at System.Threading.QueueUserWorkItemCallback.WaitCallback_Context (System.Object state) [0x00007] in <4f52a2659e1844fea7ab38854dcbcf68>:0 
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00071] in <4f52a2659e1844fea7ab38854dcbcf68>:0 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00000] in <4f52a2659e1844fea7ab38854dcbcf68>:0 
  at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem () [0x00021] in <4f52a2659e1844fea7ab38854dcbcf68>:0 
  at System.Threading.ThreadPoolWorkQueue.Dispatch () [0x00074] in <4f52a2659e1844fea7ab38854dcbcf68>:0 
  at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback () [0x00000] in <4f52a2659e1844fea7ab38854dcbcf68>:0  
exception inside UnhandledException handler: (null) assembly:/usr/lib/mono/4.5/mscorlib.dll type:TargetInvocationException member:(null)

133362 [1] ERROR CKAN.ErrorDialog (null) - Unhandled exception:
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.InvalidOperationException: The form is already displayed as a modal dialog.
  at System.Windows.Forms.Form.ShowDialog (System.Windows.Forms.IWin32Window owner) [0x00081] in <f73fa304d089422f9e23b25394111843>:0 
  at System.Windows.Forms.Form.ShowDialog () [0x00000] in <f73fa304d089422f9e23b25394111843>:0 
  at CKAN.ErrorDialog.ShowErrorDialog (System.String text, System.Object[] args) [0x00071] in <ea98fd2118b84fc7ad1a2141a259ca95>:0 
  at CKAN.ErrorDialog+<>c__DisplayClass2_0.<ShowErrorDialog>b__0 () [0x00000] in <ea98fd2118b84fc7ad1a2141a259ca95>:0 
  at (wrapper managed-to-native) System.Reflection.RuntimeMethodInfo.InternalInvoke(System.Reflection.RuntimeMethodInfo,object,object[],System.Exception&)
  at System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x0006a] in <4f52a2659e1844fea7ab38854dcbcf68>:0 
   --- End of inner exception stack trace ---
  at System.Windows.Forms.AsyncMethodResult.EndInvoke () [0x0004c] in <f73fa304d089422f9e23b25394111843>:0 
  at System.Windows.Forms.Control.EndInvoke (System.IAsyncResult asyncResult) [0x00000] in <f73fa304d089422f9e23b25394111843>:0 
  at (wrapper remoting-invoke-with-check) System.Windows.Forms.Control.EndInvoke(System.IAsyncResult)
  at System.Windows.Forms.Control.Invoke (System.Delegate method, System.Object[] args) [0x00021] in <f73fa304d089422f9e23b25394111843>:0 
  at (wrapper remoting-invoke-with-check) System.Windows.Forms.Control.Invoke(System.Delegate,object[])
  at System.Windows.Forms.Control.Invoke (System.Delegate method) [0x0001d] in <f73fa304d089422f9e23b25394111843>:0 
  at (wrapper remoting-invoke-with-check) System.Windows.Forms.Control.Invoke(System.Delegate)
  at CKAN.ErrorDialog.ShowErrorDialog (System.String text, System.Object[] args) [0x0002c] in <ea98fd2118b84fc7ad1a2141a259ca95>:0 
  at (wrapper remoting-invoke-with-check) CKAN.ErrorDialog.ShowErrorDialog(string,object[])
  at CKAN.Main.ErrorDialog (System.String text, System.Object[] args) [0x0000d] in <ea98fd2118b84fc7ad1a2141a259ca95>:0 
  at (wrapper remoting-invoke-with-check) CKAN.Main.ErrorDialog(string,object[])
  at CKAN.GUIUser.RaiseError (System.String message, System.Object[] args) [0x00005] in <ea98fd2118b84fc7ad1a2141a259ca95>:0 
  at CKAN.NetAsyncModulesDownloader.ModuleDownloadComplete (System.Uri url, System.String filename, System.Exception error) [0x0001c] in <ea98fd2118b84fc7ad1a2141a259ca95>:0 
  at CKAN.NetAsyncDownloader.FileDownloadComplete (System.Int32 index, System.Exception error) [0x00121] in <ea98fd2118b84fc7ad1a2141a259ca95>:0 
  at CKAN.NetAsyncDownloader+<>c__DisplayClass15_0.<DownloadNative>b__1 (System.Object sender, System.ComponentModel.AsyncCompletedEventArgs args) [0x00012] in <ea98fd2118b84fc7ad1a2141a259ca95>:0 
  at CKAN.NetAsyncDownloader+NetAsyncDownloaderDownloadPart.<ResetAgent>b__21_1 (System.Object sender, System.ComponentModel.AsyncCompletedEventArgs args) [0x00008] in <ea98fd2118b84fc7ad1a2141a259ca95>:0 
  at System.Net.WebClient.OnDownloadFileCompleted (System.ComponentModel.AsyncCompletedEventArgs e) [0x0000a] in <883cd000faa842148a99b174e3bd5ca4>:0 
  at System.Net.WebClient.<StartAsyncOperation>b__78_4 (System.Object arg) [0x00000] in <883cd000faa842148a99b174e3bd5ca4>:0 
  at System.Threading.QueueUserWorkItemCallback.WaitCallback_Context (System.Object state) [0x00007] in <4f52a2659e1844fea7ab38854dcbcf68>:0 
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00071] in <4f52a2659e1844fea7ab38854dcbcf68>:0 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00000] in <4f52a2659e1844fea7ab38854dcbcf68>:0 
  at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem () [0x00021] in <4f52a2659e1844fea7ab38854dcbcf68>:0 
  at System.Threading.ThreadPoolWorkQueue.Dispatch () [0x00074] in <4f52a2659e1844fea7ab38854dcbcf68>:0 
  at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback () [0x00000] in <4f52a2659e1844fea7ab38854dcbcf68>:0  
exception inside UnhandledException handler: (null) assembly:/usr/lib/mono/4.5/mscorlib.dll type:TargetInvocationException member:(null)

133363 [1] ERROR CKAN.ErrorDialog (null) - Unhandled exception:
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.InvalidOperationException: The form is already displayed as a modal dialog.
  at System.Windows.Forms.Form.ShowDialog (System.Windows.Forms.IWin32Window owner) [0x00081] in <f73fa304d089422f9e23b25394111843>:0 
  at System.Windows.Forms.Form.ShowDialog () [0x00000] in <f73fa304d089422f9e23b25394111843>:0 
  at CKAN.ErrorDialog.ShowErrorDialog (System.String text, System.Object[] args) [0x00071] in <ea98fd2118b84fc7ad1a2141a259ca95>:0 
  at CKAN.ErrorDialog+<>c__DisplayClass2_0.<ShowErrorDialog>b__0 () [0x00000] in <ea98fd2118b84fc7ad1a2141a259ca95>:0 
  at (wrapper managed-to-native) System.Reflection.RuntimeMethodInfo.InternalInvoke(System.Reflection.RuntimeMethodInfo,object,object[],System.Exception&)
  at System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x0006a] in <4f52a2659e1844fea7ab38854dcbcf68>:0 
   --- End of inner exception stack trace ---
  at System.Windows.Forms.AsyncMethodResult.EndInvoke () [0x0004c] in <f73fa304d089422f9e23b25394111843>:0 
  at System.Windows.Forms.Control.EndInvoke (System.IAsyncResult asyncResult) [0x00000] in <f73fa304d089422f9e23b25394111843>:0 
  at (wrapper remoting-invoke-with-check) System.Windows.Forms.Control.EndInvoke(System.IAsyncResult)
  at System.Windows.Forms.Control.Invoke (System.Delegate method, System.Object[] args) [0x00021] in <f73fa304d089422f9e23b25394111843>:0 
  at (wrapper remoting-invoke-with-check) System.Windows.Forms.Control.Invoke(System.Delegate,object[])
  at System.Windows.Forms.Control.Invoke (System.Delegate method) [0x0001d] in <f73fa304d089422f9e23b25394111843>:0 
  at (wrapper remoting-invoke-with-check) System.Windows.Forms.Control.Invoke(System.Delegate)
  at CKAN.ErrorDialog.ShowErrorDialog (System.String text, System.Object[] args) [0x0002c] in <ea98fd2118b84fc7ad1a2141a259ca95>:0 
  at (wrapper remoting-invoke-with-check) CKAN.ErrorDialog.ShowErrorDialog(string,object[])
  at CKAN.Main.ErrorDialog (System.String text, System.Object[] args) [0x0000d] in <ea98fd2118b84fc7ad1a2141a259ca95>:0 
  at (wrapper remoting-invoke-with-check) CKAN.Main.ErrorDialog(string,object[])
  at CKAN.GUIUser.RaiseError (System.String message, System.Object[] args) [0x00005] in <ea98fd2118b84fc7ad1a2141a259ca95>:0 
  at CKAN.NetAsyncModulesDownloader.ModuleDownloadComplete (System.Uri url, System.String filename, System.Exception error) [0x0001c] in <ea98fd2118b84fc7ad1a2141a259ca95>:0 
  at CKAN.NetAsyncDownloader.FileDownloadComplete (System.Int32 index, System.Exception error) [0x00121] in <ea98fd2118b84fc7ad1a2141a259ca95>:0 
  at CKAN.NetAsyncDownloader+<>c__DisplayClass15_0.<DownloadNative>b__1 (System.Object sender, System.ComponentModel.AsyncCompletedEventArgs args) [0x00012] in <ea98fd2118b84fc7ad1a2141a259ca95>:0 
  at CKAN.NetAsyncDownloader+NetAsyncDownloaderDownloadPart.<ResetAgent>b__21_1 (System.Object sender, System.ComponentModel.AsyncCompletedEventArgs args) [0x00008] in <ea98fd2118b84fc7ad1a2141a259ca95>:0 
  at System.Net.WebClient.OnDownloadFileCompleted (System.ComponentModel.AsyncCompletedEventArgs e) [0x0000a] in <883cd000faa842148a99b174e3bd5ca4>:0 
  at System.Net.WebClient.<StartAsyncOperation>b__78_4 (System.Object arg) [0x00000] in <883cd000faa842148a99b174e3bd5ca4>:0 
  at System.Threading.QueueUserWorkItemCallback.WaitCallback_Context (System.Object state) [0x00007] in <4f52a2659e1844fea7ab38854dcbcf68>:0 
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00071] in <4f52a2659e1844fea7ab38854dcbcf68>:0 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00000] in <4f52a2659e1844fea7ab38854dcbcf68>:0 
  at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem () [0x00021] in <4f52a2659e1844fea7ab38854dcbcf68>:0 
  at System.Threading.ThreadPoolWorkQueue.Dispatch () [0x00074] in <4f52a2659e1844fea7ab38854dcbcf68>:0 
  at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback () [0x00000] in <4f52a2659e1844fea7ab38854dcbcf68>:0  
[ERROR] FATAL UNHANDLED EXCEPTION: System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.InvalidOperationException: The form is already displayed as a modal dialog.
  at System.Windows.Forms.Form.ShowDialog (System.Windows.Forms.IWin32Window owner) [0x00081] in <f73fa304d089422f9e23b25394111843>:0 
  at System.Windows.Forms.Form.ShowDialog () [0x00000] in <f73fa304d089422f9e23b25394111843>:0 
  at CKAN.ErrorDialog.ShowErrorDialog (System.String text, System.Object[] args) [0x00071] in <ea98fd2118b84fc7ad1a2141a259ca95>:0 
  at CKAN.ErrorDialog+<>c__DisplayClass2_0.<ShowErrorDialog>b__0 () [0x00000] in <ea98fd2118b84fc7ad1a2141a259ca95>:0 
  at (wrapper managed-to-native) System.Reflection.RuntimeMethodInfo.InternalInvoke(System.Reflection.RuntimeMethodInfo,object,object[],System.Exception&)
  at System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x0006a] in <4f52a2659e1844fea7ab38854dcbcf68>:0 
   --- End of inner exception stack trace ---
  at System.Windows.Forms.AsyncMethodResult.EndInvoke () [0x0004c] in <f73fa304d089422f9e23b25394111843>:0 
  at System.Windows.Forms.Control.EndInvoke (System.IAsyncResult asyncResult) [0x00000] in <f73fa304d089422f9e23b25394111843>:0 
  at (wrapper remoting-invoke-with-check) System.Windows.Forms.Control.EndInvoke(System.IAsyncResult)
  at System.Windows.Forms.Control.Invoke (System.Delegate method, System.Object[] args) [0x00021] in <f73fa304d089422f9e23b25394111843>:0 
  at (wrapper remoting-invoke-with-check) System.Windows.Forms.Control.Invoke(System.Delegate,object[])
  at System.Windows.Forms.Control.Invoke (System.Delegate method) [0x0001d] in <f73fa304d089422f9e23b25394111843>:0 
  at (wrapper remoting-invoke-with-check) System.Windows.Forms.Control.Invoke(System.Delegate)
  at CKAN.ErrorDialog.ShowErrorDialog (System.String text, System.Object[] args) [0x0002c] in <ea98fd2118b84fc7ad1a2141a259ca95>:0 
  at (wrapper remoting-invoke-with-check) CKAN.ErrorDialog.ShowErrorDialog(string,object[])
  at CKAN.Main.ErrorDialog (System.String text, System.Object[] args) [0x0000d] in <ea98fd2118b84fc7ad1a2141a259ca95>:0 
  at (wrapper remoting-invoke-with-check) CKAN.Main.ErrorDialog(string,object[])
  at CKAN.GUIUser.RaiseError (System.String message, System.Object[] args) [0x00005] in <ea98fd2118b84fc7ad1a2141a259ca95>:0 
  at CKAN.NetAsyncModulesDownloader.ModuleDownloadComplete (System.Uri url, System.String filename, System.Exception error) [0x0001c] in <ea98fd2118b84fc7ad1a2141a259ca95>:0 
  at CKAN.NetAsyncDownloader.FileDownloadComplete (System.Int32 index, System.Exception error) [0x00121] in <ea98fd2118b84fc7ad1a2141a259ca95>:0 
  at CKAN.NetAsyncDownloader+<>c__DisplayClass15_0.<DownloadNative>b__1 (System.Object sender, System.ComponentModel.AsyncCompletedEventArgs args) [0x00012] in <ea98fd2118b84fc7ad1a2141a259ca95>:0 
  at CKAN.NetAsyncDownloader+NetAsyncDownloaderDownloadPart.<ResetAgent>b__21_1 (System.Object sender, System.ComponentModel.AsyncCompletedEventArgs args) [0x00008] in <ea98fd2118b84fc7ad1a2141a259ca95>:0 
  at System.Net.WebClient.OnDownloadFileCompleted (System.ComponentModel.AsyncCompletedEventArgs e) [0x0000a] in <883cd000faa842148a99b174e3bd5ca4>:0 
  at System.Net.WebClient.<StartAsyncOperation>b__78_4 (System.Object arg) [0x00000] in <883cd000faa842148a99b174e3bd5ca4>:0 
  at System.Threading.QueueUserWorkItemCallback.WaitCallback_Context (System.Object state) [0x00007] in <4f52a2659e1844fea7ab38854dcbcf68>:0 
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00071] in <4f52a2659e1844fea7ab38854dcbcf68>:0 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00000] in <4f52a2659e1844fea7ab38854dcbcf68>:0 
  at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem () [0x00021] in <4f52a2659e1844fea7ab38854dcbcf68>:0 
  at System.Threading.ThreadPoolWorkQueue.Dispatch () [0x00074] in <4f52a2659e1844fea7ab38854dcbcf68>:0 
  at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback () [0x00000] in <4f52a2659e1844fea7ab38854dcbcf68>:0 
exception inside UnhandledException handler: (null) assembly:/usr/lib/mono/4.5/mscorlib.dll type:TargetInvocationException member:(null)

[ERROR] FATAL UNHANDLED EXCEPTION: System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.InvalidOperationException: The form is already displayed as a modal dialog.
  at System.Windows.Forms.Form.ShowDialog (System.Windows.Forms.IWin32Window owner) [0x00081] in <f73fa304d089422f9e23b25394111843>:0 
  at System.Windows.Forms.Form.ShowDialog () [0x00000] in <f73fa304d089422f9e23b25394111843>:0 
  at CKAN.ErrorDialog.ShowErrorDialog (System.String text, System.Object[] args) [0x00071] in <ea98fd2118b84fc7ad1a2141a259ca95>:0 
  at CKAN.ErrorDialog+<>c__DisplayClass2_0.<ShowErrorDialog>b__0 () [0x00000] in <ea98fd2118b84fc7ad1a2141a259ca95>:0 
  at (wrapper managed-to-native) System.Reflection.RuntimeMethodInfo.InternalInvoke(System.Reflection.RuntimeMethodInfo,object,object[],System.Exception&)
  at System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x0006a] in <4f52a2659e1844fea7ab38854dcbcf68>:0 
   --- End of inner exception stack trace ---
  at System.Windows.Forms.AsyncMethodResult.EndInvoke () [0x0004c] in <f73fa304d089422f9e23b25394111843>:0 
  at System.Windows.Forms.Control.EndInvoke (System.IAsyncResult asyncResult) [0x00000] in <f73fa304d089422f9e23b25394111843>:0 
  at (wrapper remoting-invoke-with-check) System.Windows.Forms.Control.EndInvoke(System.IAsyncResult)
  at System.Windows.Forms.Control.Invoke (System.Delegate method, System.Object[] args) [0x00021] in <f73fa304d089422f9e23b25394111843>:0 
  at (wrapper remoting-invoke-with-check) System.Windows.Forms.Control.Invoke(System.Delegate,object[])
  at System.Windows.Forms.Control.Invoke (System.Delegate method) [0x0001d] in <f73fa304d089422f9e23b25394111843>:0 
  at (wrapper remoting-invoke-with-check) System.Windows.Forms.Control.Invoke(System.Delegate)
  at CKAN.ErrorDialog.ShowErrorDialog (System.String text, System.Object[] args) [0x0002c] in <ea98fd2118b84fc7ad1a2141a259ca95>:0 
  at (wrapper remoting-invoke-with-check) CKAN.ErrorDialog.ShowErrorDialog(string,object[])
  at CKAN.Main.ErrorDialog (System.String text, System.Object[] args) [0x0000d] in <ea98fd2118b84fc7ad1a2141a259ca95>:0 
  at (wrapper remoting-invoke-with-check) CKAN.Main.ErrorDialog(string,object[])
  at CKAN.GUIUser.RaiseError (System.String message, System.Object[] args) [0x00005] in <ea98fd2118b84fc7ad1a2141a259ca95>:0 
  at CKAN.NetAsyncModulesDownloader.ModuleDownloadComplete (System.Uri url, System.String filename, System.Exception error) [0x0001c] in <ea98fd2118b84fc7ad1a2141a259ca95>:0 
  at CKAN.NetAsyncDownloader.FileDownloadComplete (System.Int32 index, System.Exception error) [0x00121] in <ea98fd2118b84fc7ad1a2141a259ca95>:0 
  at CKAN.NetAsyncDownloader+<>c__DisplayClass15_0.<DownloadNative>b__1 (System.Object sender, System.ComponentModel.AsyncCompletedEventArgs args) [0x00012] in <ea98fd2118b84fc7ad1a2141a259ca95>:0 
  at CKAN.NetAsyncDownloader+NetAsyncDownloaderDownloadPart.<ResetAgent>b__21_1 (System.Object sender, System.ComponentModel.AsyncCompletedEventArgs args) [0x00008] in <ea98fd2118b84fc7ad1a2141a259ca95>:0 
  at System.Net.WebClient.OnDownloadFileCompleted (System.ComponentModel.AsyncCompletedEventArgs e) [0x0000a] in <883cd000faa842148a99b174e3bd5ca4>:0 
  at System.Net.WebClient.<StartAsyncOperation>b__78_4 (System.Object arg) [0x00000] in <883cd000faa842148a99b174e3bd5ca4>:0 
  at System.Threading.QueueUserWorkItemCallback.WaitCallback_Context (System.Object state) [0x00007] in <4f52a2659e1844fea7ab38854dcbcf68>:0 
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00071] in <4f52a2659e1844fea7ab38854dcbcf68>:0 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00000] in <4f52a2659e1844fea7ab38854dcbcf68>:0 
  at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem () [0x00021] in <4f52a2659e1844fea7ab38854dcbcf68>:0 
  at System.Threading.ThreadPoolWorkQueue.Dispatch () [0x00074] in <4f52a2659e1844fea7ab38854dcbcf68>:0 
  at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback () [0x00000] in <4f52a2659e1844fea7ab38854dcbcf68>:0 
133363 [1] ERROR CKAN.ErrorDialog (null) - Unhandled exception:
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.InvalidOperationException: The form is already displayed as a modal dialog.
  at System.Windows.Forms.Form.ShowDialog (System.Windows.Forms.IWin32Window owner) [0x00081] in <f73fa304d089422f9e23b25394111843>:0 
  at System.Windows.Forms.Form.ShowDialog () [0x00000] in <f73fa304d089422f9e23b25394111843>:0 
  at CKAN.ErrorDialog.ShowErrorDialog (System.String text, System.Object[] args) [0x00071] in <ea98fd2118b84fc7ad1a2141a259ca95>:0 
  at CKAN.ErrorDialog+<>c__DisplayClass2_0.<ShowErrorDialog>b__0 () [0x00000] in <ea98fd2118b84fc7ad1a2141a259ca95>:0 
  at (wrapper managed-to-native) System.Reflection.RuntimeMethodInfo.InternalInvoke(System.Reflection.RuntimeMethodInfo,object,object[],System.Exception&)
  at System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x0006a] in <4f52a2659e1844fea7ab38854dcbcf68>:0 
   --- End of inner exception stack trace ---
  at System.Windows.Forms.AsyncMethodResult.EndInvoke () [0x0004c] in <f73fa304d089422f9e23b25394111843>:0 
  at System.Windows.Forms.Control.EndInvoke (System.IAsyncResult asyncResult) [0x00000] in <f73fa304d089422f9e23b25394111843>:0 
  at (wrapper remoting-invoke-with-check) System.Windows.Forms.Control.EndInvoke(System.IAsyncResult)
  at System.Windows.Forms.Control.Invoke (System.Delegate method, System.Object[] args) [0x00021] in <f73fa304d089422f9e23b25394111843>:0 
  at (wrapper remoting-invoke-with-check) System.Windows.Forms.Control.Invoke(System.Delegate,object[])
  at System.Windows.Forms.Control.Invoke (System.Delegate method) [0x0001d] in <f73fa304d089422f9e23b25394111843>:0 
  at (wrapper remoting-invoke-with-check) System.Windows.Forms.Control.Invoke(System.Delegate)

~ [255] $  # 255 is the return code of the ckan invocation, but i guess that won't help you, looks too generic to me
```

Of course, i will end this bug report by thanking you for your work.
I would like to help, but i'm afraid C# is really too alien to me. If you have requests i could assist with, i will do my best to answer them.

--- a/GUI/ErrorDialog.Designer.cs
+++ b/GUI/ErrorDialog.Designer.cs
@@ -38,10 +38,13 @@
             //
             // panel1
             //
+            this.panel1.Anchor = ((System.Windows.Forms.AnchorStyles)((((System.Windows.Forms.AnchorStyles.Top | System.Windows.Forms.AnchorStyles.Bottom)
+            | System.Windows.Forms.AnchorStyles.Left)
+            | System.Windows.Forms.AnchorStyles.Right)));
             this.panel1.Controls.Add(this.ErrorMessage);
             this.panel1.Location = new System.Drawing.Point(13, 13);
             this.panel1.Name = "panel1";
-            this.panel1.Size = new System.Drawing.Size(267, 117);
+            this.panel1.Size = new System.Drawing.Size(775, 105);
             this.panel1.TabIndex = 0;
             //
             // ErrorMessage
@@ -50,15 +53,16 @@
             this.ErrorMessage.Dock = System.Windows.Forms.DockStyle.Fill;
             this.ErrorMessage.Location = new System.Drawing.Point(0, 0);
             this.ErrorMessage.Name = "ErrorMessage";
-            this.ErrorMessage.Size = new System.Drawing.Size(267, 117);
+            this.ErrorMessage.Size = new System.Drawing.Size(267, 105);
             this.ErrorMessage.TabIndex = 0;
             this.ErrorMessage.ReadOnly = true;
             resources.ApplyResources(this.ErrorMessage, "ErrorMessage");
             //
             // DismissButton
             //
+            this.DismissButton.Anchor = ((System.Windows.Forms.AnchorStyles)(System.Windows.Forms.AnchorStyles.Bottom));
             this.DismissButton.FlatStyle = System.Windows.Forms.FlatStyle.Flat;
-            this.DismissButton.Location = new System.Drawing.Point(205, 136);
+            this.DismissButton.Location = new System.Drawing.Point(387, 124);
             this.DismissButton.Name = "DismissButton";
             this.DismissButton.Size = new System.Drawing.Size(75, 23);
             this.DismissButton.TabIndex = 1;
@@ -70,13 +74,14 @@
             //
             this.AutoScaleDimensions = new System.Drawing.SizeF(6F, 13F);
             this.AutoScaleMode = System.Windows.Forms.AutoScaleMode.Font;
-            this.ClientSize = new System.Drawing.Size(292, 174);
-            this.ControlBox = false;
+            this.ClientSize = new System.Drawing.Size(800, 160);
+            this.ControlBox = true;
             this.Controls.Add(this.DismissButton);
             this.Controls.Add(this.panel1);
-            this.FormBorderStyle = System.Windows.Forms.FormBorderStyle.FixedDialog;
+            this.FormBorderStyle = System.Windows.Forms.FormBorderStyle.SizableToolWindow;
             this.Icon = Properties.Resources.AppIcon;
             this.Name = "ErrorDialog";
+            this.StartPosition = System.Windows.Forms.FormStartPosition.CenterParent;
             resources.ApplyResources(this, "$this");
             this.panel1.ResumeLayout(false);
             this.ResumeLayout(false);

--- a/GUI/ErrorDialog.cs
+++ b/GUI/ErrorDialog.cs
@@ -1,33 +1,42 @@
 ﻿using System;
+using System.Drawing;
 using System.Windows.Forms;
 using log4net;
 
 namespace CKAN
 {
-
-    public partial class ErrorDialog : FormCompatibility
+    public partial class ErrorDialog : Form
     {
-        private static readonly ILog log = LogManager.GetLogger(typeof(ErrorDialog));
-
         public ErrorDialog()
         {
             InitializeComponent();
-            ApplyFormCompatibilityFixes();
-            StartPosition = FormStartPosition.CenterScreen;
         }
 
         public void ShowErrorDialog(string text, params object[] args)
         {
-            if (Main.Instance.InvokeRequired)
-            {
-                Main.Instance.Invoke(new MethodInvoker(() => ShowErrorDialog(text, args)));
-            }
-            else
+            Util.Invoke(Main.Instance, () =>
             {
                 log.ErrorFormat(text, args);
-                ErrorMessage.Text = String.Format(text, args);
-                ShowDialog();
-            }
+                // Append to previous text, if any
+                if (!string.IsNullOrEmpty(ErrorMessage.Text))
+                {
+                    ErrorMessage.Text += Environment.NewLine + Environment.NewLine;
+                }
+                ErrorMessage.Text += string.Format(text, args);
+                // Resize form to fit
+                var padding = ClientSize.Height - ErrorMessage.Height + 50;
+                ClientSize = new Size(
+                    ClientSize.Width,
+                    Math.Min(
+                        maxHeight,
+                        padding + StringHeight(ErrorMessage.Text, ErrorMessage.Width - 4)
+                    )
+                );
+                if (!Visible)
+                {
+                    ShowDialog(Main.Instance);
+                }
+            });
         }
 
         public void HideErrorDialog()
@@ -39,5 +48,19 @@ namespace CKAN
         {
             Close();
         }
+
+        protected override void OnClosed(EventArgs e)
+        {
+            // Clear message on close so we start blank next time
+            ErrorMessage.Text = "";
+        }
+
+        private int StringHeight(string text, int maxWidth)
+        {
+            return (int)CreateGraphics().MeasureString(text, ErrorMessage.Font, maxWidth).Height;
+        }
+
+        private const           int  maxHeight = 600;
+        private static readonly ILog log       = LogManager.GetLogger(typeof(ErrorDialog));
     }
 }

--- a/GUI/ErrorDialog.resx
+++ b/GUI/ErrorDialog.resx
@@ -118,7 +118,6 @@
     <value>System.Resources.ResXResourceWriter, System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
   </resheader>
   <assembly alias="System.Drawing" name="System.Drawing, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" />
-  <data name="ErrorMessage.Text" xml:space="preserve"><value>Error!</value></data>
   <data name="DismissButton.Text" xml:space="preserve"><value>Dismiss</value></data>
   <data name="$this.Text" xml:space="preserve"><value>Error!</value></data>
 </root>

--- a/GUI/Localization/de-DE/ErrorDialog.de-DE.resx
+++ b/GUI/Localization/de-DE/ErrorDialog.de-DE.resx
@@ -118,7 +118,6 @@
     <value>System.Resources.ResXResourceWriter, System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
   </resheader>
   <assembly alias="System.Drawing" name="System.Drawing, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" />
-  <data name="ErrorMessage.Text" xml:space="preserve"><value>Fehler!</value></data>
   <data name="DismissButton.Text" xml:space="preserve"><value>OK</value></data>
   <data name="$this.Text" xml:space="preserve"><value>Fehler!</value></data>
 </root>

